### PR TITLE
fix: ignore *.d.ts files for migration

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes  [UNIFY-1219](https://cypress-io.atlassian.net/browse/UNIFY-1219)

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: system-tests/projects/migration-e2e-component-default-with-types/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+| `integration/spec.ts` | `e2e/spec.cy.ts` |
+| `component/button.spec.js` | `component/button.cy.js` |
+
+## Manual Files
+
+The user will need to rename and/or move their only component spec, `button.cy.js` to its new location.
+
+## Rename supportFile
+
+The project has a default support file, `cypress/support/index.js`. We can rename it for them to `cypress/support/e2e.js`. There is no such `cypress/support/component.js` created here - that's going to part of the componen ttesting reconfiguration workflow.
```
Comment (lmiller1990) [inline system-tests/projects/migration-e2e-component-default-with-types/README.md:29]:
```suggestion
The project has a default support file, `cypress/support/index.js`. We can rename it for them to `cypress/support/e2e.js`. There is no such `cypress/support/component.js` created here - that's going to part of the component testing reconfiguration workflow.
```

File: packages/data-context/src/sources/migration/codegen.ts [hunk 2]
```diff
@@ -261,13 +261,13 @@ export async function hasSpecFile (projectRoot: string, folder: string, glob: st
...
 export async function tryGetDefaultLegacyPluginsFile (projectRoot: string) {
-  const files = await globby('cypress/plugins/index.*', { cwd: projectRoot })
+  const files = await globby('cypress/plugins/index.*', { cwd: projectRoot, ignore: ['cypress/plugins/index.d.*'] })
 
   return files[0]
 }
 
 export async function tryGetDefaultLegacySupportFile (projectRoot: string) {
-  const files = await globby('cypress/support/index.*', { cwd: projectRoot })
+  const files = await globby('cypress/support/index.*', { cwd: projectRoot, ignore: ['cypress/support/index.d.*'] })
```
Comment (lmiller1990) [inline packages/data-context/src/sources/migration/codegen.ts:270]: Should this be `index.d.ts`? Why `index.d.*`?

File: packages/data-context/src/sources/migration/codegen.ts [hunk 3]
```diff
@@ -261,13 +261,13 @@ export async function hasSpecFile (projectRoot: string, folder: string, glob: st
 }
 
 export async function tryGetDefaultLegacyPluginsFile (projectRoot: string) {
-  const files = await globby('cypress/plugins/index.*', { cwd: projectRoot })
+  const files = await globby('cypress/plugins/index.*', { cwd: projectRoot, ignore: ['cypress/plugins/index.d.*'] })
```
Comment (emilyrohrbough) [inline packages/data-context/src/sources/migration/codegen.ts]: Changes look fine. Just curious though, why are we globbing this? We know where the the legacy configuration file is located and could pull this value from there and/or fallback to the previously file path.

Comment (ImCesar) [reply packages/data-context/src/sources/migration/codegen.ts]: We know where it's located by default, but the default extension could be either `index.js`, `index.ts`, or `index.coffee`. So globbing it makes it easy to see if any of those are there.

**Review by emilyrohrbough — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**